library/accessory_workflows/build_cellranger_reference/BuildCellRangerReferenceBundle.wdl

```diff
@@ -0,0 +1,102 @@
+
+  output {
+    File cellranger_ref = BuildCellRangerReference.cellRangerRef
+  }
+}
+
+task BuildCellRangerReference {
+  input {
+    String ref_fasta
+    String gtf_file
+    String ref_name
+
+    String docker = "quay.io/humancellatlas/secondary-analysis-cellranger:v1.0.0"
+    String memory = "57 GB"
+    Int boot_disk_size_gb = 12
+    Int disk_space = 50
+    Int cpu = 64
+  }
+  parameter_meta {
+    ref_fasta: "Genome reference in fasta format"
+    gtf_file: "Gene annotation file in gtf format"
+    disk_space: "(optional) the amount of disk space (GB) to provision for this task"
+    ref_name: "The expected file name of the output reference"
+    docker: "(optional) the docker image containing the runtime environment for this task"
+    boot_disk_size_gb: "(optional) the amount of space (GB) of boot disk to provision for this task"
+    cpu: "(optional) the number of cpus to provision for this task"
+    memory: "(optional) the amount of memory (MB) to provision for this task"
+  }
+
+  command <<<
+
+    set -euo pipefail
+
+    ref_fasta_filename="$(basename "~{ref_fasta}")"
+    ref_fasta_filename="${ref_fasta_filename%.gz}"
+    wget -q -O - "~{ref_fasta}"| zcat > "$ref_fasta_filename"
+
+    wget ~{gtf_file}
+    gtf_filename="$(basename "~{gtf_file}")"
+    gtf_filename="${gtf_filename%.gz}"
+    wget -q -O - "~{gtf_file}"| zcat > "$gtf_filename"
+
+    # The reference building process follows the instructions from
+    # https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/references
+
+    cellranger mkgtf "$gtf_filename" "filtered.$gtf_filename" \
+                     --attribute=gene_biotype:protein_coding \
+                     --attribute=gene_biotype:lincRNA \
+                     --attribute=gene_biotype:antisense \
+                     --attribute=gene_biotype:IG_LV_gene \
+                     --attribute=gene_biotype:IG_V_gene \
+                     --attribute=gene_biotype:IG_V_pseudogene \
+                     --attribute=gene_biotype:IG_D_gene \
+                     --attribute=gene_biotype:IG_J_gene \
+                     --attribute=gene_biotype:IG_J_pseudogene \
+                     --attribute=gene_biotype:IG_C_gene \
+                     --attribute=gene_biotype:IG_C_pseudogene \
+                     --attribute=gene_biotype:TR_V_gene \
+                     --attribute=gene_biotype:TR_V_pseudogene \
+                     --attribute=gene_biotype:TR_D_gene \
+                     --attribute=gene_biotype:TR_J_gene \
+                     --attribute=gene_biotype:TR_J_pseudogene \
+                     --attribute=gene_biotype:TR_C_gene
+
+
+    cellranger mkref --genome=GRCh38 \
+                     --fasta="$ref_fasta_filename" \
+                     --genes="filtered.$gtf_filename" \
+                     --nthreads=$(getconf _NPROCESSORS_ONLN)
+
+    tar cvf "~{ref_name}.tar" GRCh38/
+
+  >>>
+
+  runtime {
+    bootDiskSizeGb: boot_disk_size_gb
```

Comment:
Why are we specifying this here? We don't do it in other pipelines, do we?

Reply:
We haven't done this in other pipelines, except the cellranger one: https://github.com/HumanCellAtlas/skylab/blob/ffd17882c831da377dce3ab471c7a44683dd00f6/pipelines/cellranger/cellranger.wdl#L16 Don't know if it's a required, but just copied over

Reply:
@samanehsan can you comment on why the cellranger pipeline specifies this?

Reply:
@dshiga this was a part of the cellranger.wdl that was in firecloud, and I kept the same parameters

Reply:
@dshiga I found it require a lot of space on our boot disk(even you have a large mounted disk) when you pull the cellranger docker image from quay.io The docker daemon will complain and exit unexpectedly. So I prefer we keep this parameter here to provide some flexibility to override it from input json files.

Reply:
Great, thanks for looking into it @rexwangcc. Would you mind adding a code comment explaining that this parameter is here because the cellranger image is so large?

Reply:
Will do!

Reply:
Just for the record, I think we need to increase `bootDiskSizeGb` if the  programs(java application, for example) we run require a lot of TMP space.

Reply:
@jishuxu That makes sense! Another reason to keep it configurable here!